docs/source/fx.rst

```diff
@@ -298,6 +290,245 @@ Miscellanea
+into it manually using ``pdb`` when the forward pass is invoked.
+
+::
+
+    x, y = torch.rand(2, 3), torch.rand(2, 3)
+    m = symbolic_trace(M())
+    import pdb; pdb.set_trace()
+    m(x, y)
+
+.. _Print the Generated Code:
+
+Print the Generated Code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you’d like to run the same code multiple times, then it can be
+a bit tedious to step to the right code with ``pdb``. In that case, one
+approach is to simply copy-paste the generated ``forward`` pass into
+your code and examine it from there.
+
+::
+
+    # Module whose trace we want to debug
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.x = torch.nn.Parameter(torch.rand(4, 3))
+
+        def forward(self, y):
+            return self.x + y
+
+    # Symbolically trace an instance of `M`
+    traced = symbolic_trace(M())
+
+    # Print the code generated from symbolic tracing. This outputs:
+    """
+    def forward(self, y):
+        x = self.x
+        add_1 = x + y;  x = y = None
+        return add_1
+    """
+    # Copy this code for later
+    print(traced)
+
+    # Subclass the original Module
+    class SubclassM(M):
+        def __init__(self):
+            super().__init__()
+
+        # Paste the generated `forward` function (the one we printed and
+        # copied on line 22) here
+        def forward(self, y):
+            x = self.x
+            add_1 = x + y;  x = y = None
+            return add_1
+
+    # Create an instance of the original, untraced Module. Then, create an
+    # instance of the Module with the copied `forward` function. We can
+    # now compare the output of both the original and the traced version.
+    pre_trace = M()
+    post_trace = SubclassM()
+
+
+Use the ``to_folder`` Function From ``GraphModule``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:meth:`GraphModule.to_folder` is a method in ``GraphModule`` that allows
+you to dump out the generated FX code to a folder. Although copying the
+forward pass into the code often suffices as in :ref:`Print the Generated Code`,
+it doesn’t capture any model attribute state. To examine modules and parameters,
+we can use ``to_folder``.
+
+::
+
+    m = symbolic_trace(M())
+    m.to_folder("foo", "Bar")
+    from foo import Bar
+    y = Bar()
+
+After running the above example, we can then look at the code within
+``foo/module.py`` and modify it as desired (e.g. adding ``print``
+statements or using ``pdb``) to debug the generated code.
+
+Debugging the Transformation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we've identified that a transformation is creating incorrect
+code, it's time to debug the transformation itself. We’ve
+checked the :ref:`Limitations of Symbolic Tracing` section in the
+documentation, and we know that ``symbolic_trace`` is working as
+expected. Our goal is to figure out what went wrong during our
+transformation of the generated FX ``GraphModule``. First, we need to
```

Comment:
If the user wrote the transformation, hopefully they're familiar with these classes. Maybe it's better to emphasize reading the docs in the "Writing FX Transformations" section?